src/ansible_navigator/actions/lint.py

```diff
@@ -0,0 +1,343 @@
+from ..ui_framework import nonblocking_notification
+from ..utils import abs_user_path
+from . import _actions as actions
+from . import run_action
+
+
+class Severity(IntEnum):
+    """
+    A mapping from ansible-lint severity to an integer represented internally.
+    Primarily used for sorting.
+    """
+
+    INFO = 10
+    MINOR = 20
+    MAJOR = 30
+    CRITICAL = 40
+    BLOCKER = 50
+
+    # If ansible-lint ever gives us something we don't expect, return this.
+    # In practice, we reverse-sort, so this will always be last.
+    UNKNOWN = -1
+
+    @classmethod
+    def _missing_(cls, value):
+        """
+        If ansible-lint ever gives us something we don't expect return something
+        that tells us.
+        """
+        return Severity.UNKNOWN
+
+
+def severity_to_color(severity: str) -> int:
+    """Convert severity to curses colors."""
+    if severity == "minor":
+        return 5
+    if severity == "major":
+        return 3
+    if severity in ("critical", "blocker"):
+        return 1
+    if severity == "info":
+        return 6
+    return 0
+
+
+def color_menu(colno: int, colname: str, entry: Dict[str, Any]) -> Tuple[int, int]:
+    # pylint: disable=unused-argument
+    """Color the menu."""
+    return (severity_to_color(entry["severity"]), 0)
+
+
+def content_heading(obj: Dict, screen_w: int) -> Union[CursesLines, None]:
+    """Generate the content heading."""
+    check_name = obj["check_name"]
+    check_name = check_name + (" " * (screen_w - len(check_name)))
+    path_line = f"PATH: {abs_user_path(obj['location']['path'])}"
+    check_name_line = f"MESSAGE: {check_name}"
+
+    return (
+        (
+            CursesLinePart(
+                column=0,
+                string=path_line,
+                color=0,
+                decoration=curses.A_BOLD,
+            ),
+        ),
+        (
+            CursesLinePart(
+                column=0,
+                string=check_name_line,
+                color=severity_to_color(obj["severity"]),
+                decoration=curses.A_UNDERLINE,
+            ),
+        ),
+    )
+
+
+def filter_content_keys(obj: Dict[Any, Any]) -> Dict[Any, Any]:
+    """when showing content, filter out some keys"""
+    ignored_keys = ("fingerprint",)
+    return {k: v for k, v in obj.items() if not k.startswith("__") and k not in ignored_keys}
+
+
+def massage_issues(issues: List[Dict]) -> List[Dict]:
+    """Massage issues by injecting some useful keys with strings for rendering."""
+    out = []
+    for issue in issues:
+        issue["__message"] = issue["check_name"].split("] ", 1)[1].capitalize()
+        issue["__path"] = abs_user_path(issue["location"]["path"])
+        if isinstance(issue["location"]["lines"]["begin"], Mapping):
+            issue["__line"] = issue["location"]["lines"]["begin"]
+        else:
+            issue["__line"] = issue["location"]["lines"]["begin"]
+        issue["issue_path"] = f"{issue['__path']}:{issue['__line']}"
```

Comment:
Could you move this set of transformations into a dedicated pure function? It'd be easier to understand it this way.